src/features/courses/course-controller.ts

```diff
@@ -3083,6 +3085,24 @@ class CourseController {
         if (!_.isNil(topic.studentTopicOverride) && topic.studentTopicOverride.length > 0) {
             topic = topic.getWithOverrides(topic.studentTopicOverride[0]) as CourseTopicContent;
         }
+        const unit = await topic.getUnit();
+        const course = await unit.getCourse();
+        const enroll = await course.getEnrolledStudents({
+            where: {
+                userId: user.id,
+            }
+        });
+        if (_.isEmpty(enroll)) {
+            logger.warn(`User #${user.id} is not enrolled in course for topic #${topicId}.`);
```

Comment:
I don't think this warning should be here
This is not an error in anyway on the backend

The frontend should report the error since it got to a place it shouldn't have